src/AppInstallerCommonCore/Fonts.cpp

```diff
@@ -44,20 +67,647 @@ namespace AppInstaller::Fonts
+                        {
+                            continue;
+                        }
+                        const auto& valueName = ConvertToUTF16(valueRef.Name());
+                        const std::filesystem::path& valuePath = { value->GetValue<Registry::Value::Type::String>() };
+                        if (valuePath.is_relative())
+                        {
+                            // Relative value is OK, just compare the filenames.
+                            // We will assume this is in the standard font paths.
+                            if (valuePath.filename() == filePath.filename())
+                            {
+                                return valueName;
+                            }
+                        }
+                        else
+                        {
+                            if (valuePath == filePath)
+                            {
+                                return valueName;
+                            }
+                        }
+                    }
+                }
+            }
+            CATCH_LOG();
+
+            return std::nullopt;
+        }
+    }
+
+    FontResult FontOperationResult::Result()
+    {
+        if (FAILED(HResult))
+        {
+            return FontResult::Error;
+        }
+        else
+        {
+            return FontResult::Success;
+        }
+    }
+
+    void FontContext::AddPackageFile(const std::filesystem::path& filePath)
+    {
+        if (!PackageFiles.has_value())
+        {
+            PackageFiles = std::vector<std::filesystem::path>();
+        }
+
+        PackageFiles.value().push_back(filePath);
+    }
+
+    std::wstring GetFontRegistryPath(const FontContext& context)
+    {
+        // Registry path for a font is well-defined based on installer source, and if present a package name.
+        auto path = std::wstringstream();
+        path << s_FontsPathSubkey;
+
+        switch (context.InstallerSource)
+        {
+        case InstallerSource::Unknown:
+            // Unknown installer is assumed to be installed in the default locations.
+            break;
+        case InstallerSource::UWP:
+            // UWP path adds the package full name
+            if (context.PackageIdentifier.has_value() && !context.PackageIdentifier.value().empty())
+            {
+                path << s_RegistrySeparator << context.PackageIdentifier.value();
+            }
+            break;
+        case InstallerSource::WinGet:
+            // WinGet path adds the WinGet prefix and the package name
+            if (context.PackageIdentifier.has_value() && !context.PackageIdentifier.value().empty())
+            {
+                path << s_RegistrySeparator << s_FontsWinGetPrefix << s_RegistrySeparator << context.PackageIdentifier.value();
+            }
+            else
+            {
+                throw std::runtime_error("WinGet Font paths require a package name.");
```

Comment:
If you use std exceptions, you will have to put a catch-wall at the front-end so that they can be presented to the user properly (localized messages).  Prefer `THROW_HR` with our errors.  When the front-end catches those it will output the localized version of the error message automatically.

The exception (PUN :100:) to that rule would be if these are all logic errors that you only expect to occur due to bugs, not based on user or data malformities.